components/cam/src/physics/cam/scream_abortutils.F90

```diff
@@ -0,0 +1,51 @@
+module scream_abortutils
+
+!-------------------------------------------------
+!Utilities to stop the model in case of
+!catastrophic errors
+!-------------------------------------------------
+implicit none
+private
+
+!public subroutines
+public :: endscreamrun
+
+contains
+
+  subroutine endscreamrun (msg)
+    !-------------------------------------------------
+    ! This subroutine will print the optional message
+    ! received via optional arg "msg" and stop
+    ! the simulation
+    !-------------------------------------------------
+
+#ifdef SPMD
+    !Modules to use when SCREAM run with MPI under E3SM
+    use micro_p3_utils, only:iulog_e3sm
+    use cam_abortutils, only: endrun
```

Comment:
So, this pulls in a lot of stuff from `cime/src/share/util`, right? What if we simply did

```
#ifdef SPMD
#include <mpif.h>
    integer :: ierr
#endif
   if(present(msg)) write(*,*) trim(adjust(msg))
   
#ifdef SPMD
   call MPI_Abort(MPI_COMM_WORLD,-1,ierr)
#else
   call abort()
#endif
```

Reply:
I thought about implementing this way but then I thought I can use the existing E3SM mechanism as the MPI abort is only needed when scream is run within E3SM. Your suggested approach is more straightforward and eliminates E3SM dependencies.

Reply:
The problem is that if this file is compiled also in standalone scream (and right now it is, given how you added [here](https://github.com/E3SM-Project/scream/blob/4f39ae34108f33301ba74aedf692ec5cb94213d1/components/scream/src/physics/p3/CMakeLists.txt#L15)), then adding deps on E3SM cime stuff forces us to compile that code too (which right now we don't). Either remove the E3SM deps (which is easy, I think), or make it so that SCREAM standalone (and only standalone) also adds E3SM files to the list of stuff to build.

Reply:
These dependencies are only needed when we compile SCREAM within E3SM (encapsulated in SPMD CPP, which is true only when SCREAM is run under E3SM). I have built and run both standalone SCREAM and SCREAM within E3SM. Both compilations and model runs work fine. The AT machines also ran fine with these changes.

Reply:
Ah, nice. Great then.